Userland/Libraries/LibPDF/Error.h

```diff
@@ -27,6 +28,11 @@ class Error {
     }
 
     Error(Type type, DeprecatedString const& message)
+        : Error(type, String::from_deprecated_string(message).release_value_but_fixme_should_propagate_errors())
+    {
+    }
+
+    Error(Type type, String const& message)
```

Comment:
This only needs to take a StringView. Also don't add a new constructor that does string formatting, which will no longer work when upgrading to fallible string.

Reply:
I'm not sure if I'm reading your comment right, but are you suggesting that this `Error` class shouldn't allow for formatted messages at all, and that it should only be able to receive fixed string views? (like `AK::Error`?)

Note that this `Error` class' constructor already previously did string formatting. The change I'm introducing doesn't add more formatting to the ctror, it only allows it to take an already-constructed `String` instead of a `DeprecatedString` (but I kept the one taking a `DeprecatedString`, otherwise the changeset would have been too big). The new ctor is then invoked from the new `Error::name_##_error(CheckedFormatString....)` static functions. They internally create the `String`, or an empty one if that fails.

Due to these two reasons (keep changeset small, and still allow formatted messages) it still makes sense in my mind to, at the end of the day, receive a `String` in the ctor. However, if the intention is to never allow formatted messages then I'd agree that receiving a `StringView` would be the way to go. That'd be a way bigger set of changes though.